src/components/MjmlConfigPath.js

```diff
@@ -0,0 +1,205 @@
+import fs from 'fs'
+import React, { Component } from 'react'
+import debounce from 'lodash/debounce'
+import { connect } from 'react-redux'
+import { handleMjmlConfig } from 'mjml'
+
+import IconCheck from 'react-icons/md/check-circle'
+import IconChecking from 'react-icons/md/autorenew'
+import IconError from 'react-icons/md/error'
+import IconWarning from 'react-icons/md/warning'
+
+import { updateSettings } from 'actions/settings'
+
+import Button from 'components/Button'
+// import RadioGroup from 'components/RadioGroup'
+// import Radio from 'components/RadioGroup/Radio'
+
+
+@connect(
+  state => {
+    const { settings } = state
+    return {
+      mjmlConfigPath: settings.getIn(['mjml', 'mjmlConfigPath'], ''),
+      mjmlEngine: settings.getIn(['mjml', 'engine'], 'auto'),
+    }
+  },
+  {
+    updateSettings,
+  },
+)
+class MjmlConfigPath extends Component {
+  state = {
+    mjmlConfigPath: '',
+    // unset / checking / valid / invalid / warning
+    pathStatus: 'unset',
+    message: {},
+  }
+
+  componentWillMount() {
+    const { mjmlConfigPath } = this.props
+    this.setState({
+      mjmlConfigPath,
+      pathStatus: mjmlConfigPath ? 'checking' : 'unset',
+    })
+  }
+
+  componentDidMount() {
+    const { mjmlConfigPath } = this.state
+    if (mjmlConfigPath) {
+      window.requestIdleCallback(() => {
+        this.checkMjmlConfig(mjmlConfigPath)
+      })
+    }
+  }
+
+  componentDidUpdate(prevProps) {
+    if (prevProps.mjmlEngine !== this.props.mjmlEngine) {
+      this.checkMjmlConfig(this.state.mjmlConfigPath)
+    }
+  }
+
+  componentWillUnmount() {
+    this._unmounted = true
+  }
+
+  handleChangePath = p => {
+    if (!p) {
+      return this.setState({ pathStatus: 'unset', mjmlConfigPath: '' })
+    }
+    this.setState({ mjmlConfigPath: p })
+    this.checkMjmlConfig(p)
+    this.debounceSaveSettings()
+  }
+
+  handleBrowse = () => {
+    const p = fileDialog({
+      properties: ['openFile'],
+    })
+    if (!p) {
+      return
+    }
+    this.handleChangePath(p)
+  }
+
+  formatTestResult(result) {
+    const { error, success, failures } = result
+
+    if (success) {
+      const formattedFailures = failures.map(f => {
+        if (f.error.code === 'ENOENT' || f.error.code === 'MODULE_NOT_FOUND') {
+          return `File not found or unreadable : ${f.compPath}`
+        }
+        return `Error when registering component "${f.compPath}" - ${f.error}`
+      })
+
+      return {
+        pathStatus: failures.length ? 'warning' : 'valid',
+        message: [
+          `${success.length} component(s) successfully registered:`,
+          ...(success.length ? success : ['none']),
+          failures.length ? `${failures.length} error(s):` : 'No errors during registration',
+          ...formattedFailures
+        ].join('<br/>'),
```

Comment:
prefer joining the strings with line break `\n` then instead of using `dangerouslySetInnerHTML` you can use a regular `div`/`span` with `white-space: pre`

Reply:
Actually since handleMjmlConfig is no longer exported in mjml, it's no longer possible to check the .mjmlconfig file and get the result of the components registration, so i had to remove the whole check